src/topology/sober.lean

```diff
@@ -0,0 +1,282 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import topology.separation
+import topology.continuous_function.basic
+
+/-!
+# Sober spaces
+
+A sober space is a topological space where every irreducible closed subset has a generic point.
+Note that we do not require the generic point to be unique.
```

Comment:
Usually this condition is included. And I think there are good reasons to do so. Should we call the version without uniqueness `presober`?

Reply:
This is called `quasi-sober` in stacks project.
I figured that since (the usual) sobriety can be easily stated with `[t0_space α] [sober_space α]`, I might as well use the shorter name.

However this may cause confusion I suppose?